ROX-16495: Installation instructions for PostgreSQL tech preview (#58668)
openshift/openshift-docs: modules/install-central-operator-tech-preview.adoc

// Module included in the following assemblies:
//
// * installing/install-ocp-operator.adoc
:_module-type: PROCEDURE
[id="install-central-operator-tech-preview_{context}"]
= Installing Central with an external database using the Operator method

:FeatureName: External PostgreSQL support
include::snippets/technology-preview.adoc[]

[role="_abstract"]
The main component of {product-title} is called Central. You can install Central on {ocp} by using the `Central` custom resource. You deploy Central only once, and you can monitor multiple separate clusters by using the same Central installation.

[IMPORTANT]
====
When you install {product-title} for the first time, you must first install the `Central` custom resource because the `SecuredCluster` custom resource installation is dependent on certificates that Central generates.
====

.Prerequisites
* You must be using {ocp} 4.6 or later.
* Provision a database that supports PostgreSQL 13 that will be used solely for ACS (shared infrastructure is currently unsupported).

[NOTE]
====
* A multi-tenant database is currently unsupported.
* Superuser role is required.

> * PostgreSQL 14 or 15 _may_ work, but is not officially supported.

Author: will do

* PgBouncer is currently unsupported.
====

.Procedure
. On the {ocp} web console, navigate to the *Operators* -> *Installed Operators* page.
. Select the {product-title} Operator from the list of installed Operators.
. If you have installed the Operator in the recommended namespace, {ocp} lists the project as `rhacs-operator`. Select *Project: rhacs-operator* -> *Create project*.

> [WARNING]

Author: will do

====
* If you have installed the Operator in a different namespace, {ocp} shows the name of that namespace rather than `rhacs-operator`.
* You must install the {product-title} `Central` custom resource in its own project and not in the `rhacs-operator` and `openshift-operator` projects, or in the project in which you have installed the {product-title} Operator.
====
. Enter the new project name (for example, `stackrox`), and click *Create*. Red Hat recommends that you use `stackrox` as the project name.
. Using the terminal, create a password secret in the newly created project. password.txt is the password for the superuser of the provisioned database.
[source,terminal]
----
$ oc create secret generic external-db-password --from-file=password=password.txt
----
. Under the *Provided APIs* section, select *Central*. Click *Create Central*.
. Enter a name for your `Central` custom resource and add any labels you want to apply.
. Navigate to *Central Component Settings* > *Central DB Settings*.
. Underneath *Administrator Password* specify the referenced secret as `external-db-password` (or the secret name of the password created above).

> . Specify the connection string in keyword/value format as described https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING[here]. e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`

Reviewer: ```suggestion
. Underneath *Connection String (Technology Preview)* specify the connection string in keyword/value format as described https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING[here]. e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`
```

. Underneath *Persistence* > *PersistentVolumeClaim* > *Claim Name* remove `central-db`.
. If necessary, a Certificate Authority can be specified in order for Central to trust the database certificate. This can be added by navigating to the YAML view and adding a TLS block underneath the top-level spec:

> [source,yaml]
> ----

Reviewer: ```suggestion
+
[source,yaml]
----
```

// Module included in the following assemblies:
//
// * installing/install-ocp-operator.adoc
:_module-type: PROCEDURE
[id="install-central-operator-tech-preview_{context}"]

> = Installing Central with an external database using the Operator method

Reviewer 2: Installing Central with an external database using the {product-title-short} Operator

:FeatureName: External PostgreSQL support
include::snippets/technology-preview.adoc[]

[role="_abstract"]
The main component of {product-title} is called Central. You can install Central on {ocp} by using the `Central` custom resource. You deploy Central only once, and you can monitor multiple separate clusters by using the same Central installation.

[IMPORTANT]
====
When you install {product-title} for the first time, you must first install the `Central` custom resource because the `SecuredCluster` custom resource installation is dependent on certificates that Central generates.
====

.Prerequisites

> * You must be using {ocp} 4.6 or later.

Reviewer 3: This just got changed recently to 4.10 or later so you might get a merge conflict. As of ACS 4.0, it's 4.10

> * Provision a database that supports PostgreSQL 13 that will be used solely for ACS.

Reviewer 3: ```suggestion
* Provision a database that supports PostgreSQL 13 that will be used solely for {product-title-short}.
```

> * Provision a database that supports PostgreSQL 13 that will be used solely for {product-title-short}.

Reviewer 2: Create a PostgreSQL 13 database dedicated exclusively to {product-title-short}.

> [NOTE]
> ====
> * A multi-tenant database is currently unsupported.
> * Superuser role is required with permissions to create and delete databases.
> * PgBouncer is currently unsupported.
> ====
> 

Reviewer 2: ```suggestion
[NOTE]
====
{product-title-short} 4.0 does not support a multi-tenant database and PgBouncer.
====

.Prerequisites
* You must have a superuser role with permissions to create and delete databases.

```

.Procedure
. On the {ocp} web console, navigate to the *Operators* -> *Installed Operators* page.
. Select the {product-title} Operator from the list of installed Operators.
. If you have installed the Operator in the recommended namespace, {ocp} lists the project as `rhacs-operator`. Select *Project: rhacs-operator* -> *Create project*.

+
[WARNING]
====
* If you have installed the Operator in a different namespace, {ocp} shows the name of that namespace rather than `rhacs-operator`.
* You must install the {product-title} `Central` custom resource in its own project and not in the `rhacs-operator` and `openshift-operator` projects, or in the project in which you have installed the {product-title} Operator.
====
. Enter the new project name (for example, `stackrox`), and click *Create*. Red Hat recommends that you use `stackrox` as the project name.

> . Create a password secret by navigating to the *Workloads* -> *Secrets* page on the {ocp} web console. Create a `Key/Value secret` with the key `password` and the value being a plaintext file containing the password for the superuser of the provisioned database.

Reviewer 3: ```suggestion
. Create a password secret by navigating to the *Workloads* -> *Secrets* page on the {ocp} web console. Create a `Key/Value secret` with the key `password` and the value of a plaintext file containing the password for the superuser of the provisioned database.
```

> Alternatively, using the terminal, create a password secret in the newly created project. password.txt is the password for the superuser of the provisioned database.

Reviewer 3: ```suggestion
Alternatively, using the terminal, create a password secret in the newly created project. password.txt is the password for the superuser of the provisioned database. Enter the following command:
```

+
[source,terminal]
----
$ oc create secret generic external-db-password --from-file=password=password.txt
----
. Return to the {product-title} operator page in the {ocp} web console. Under the *Provided APIs* section, select *Central*. Click *Create Central*.
. Enter a name for your `Central` custom resource and add any labels you want to apply.

> . Navigate to *Central Component Settings* > *Central DB Settings*.

Reviewer 3: ```suggestion
. Navigate to *Central Component Settings* -> *Central DB Settings*.
```

Reviewer 3: (when transformed to HTML this creates an arrow)

> . Underneath *Administrator Password* specify the referenced secret as `external-db-password` (or the secret name of the password created above).

Reviewer 3: ```suggestion
. For *Administrator Password*, specify the referenced secret as `external-db-password` (or the secret name of the password created previously).
```

> . Underneath *Connection String (Technology Preview)* specify the connection string in keyword/value format as described in the PostgreSQL documentation in "Additional resources". e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`

Reviewer 3: ```suggestion
. For *Connection String (Technology Preview)*, specify the connection string in keyword/value format as described in the PostgreSQL documentation in "Additional resources".; for example, `host=<host> port=5432 user=postgres sslmode=verify-ca`
```

> . Underneath *Connection String (Technology Preview)* specify the connection string in keyword/value format as described https://www.postgresql.org/docs/current/static/libpq-connect.html#LIBPQ-CONNSTRING[here]. e.g. `host=<host> port=5432 user=postgres sslmode=verify-ca`

Reviewer 3: link issue, see my comments on https://github.com/openshift/openshift-docs/pull/58655

> . Underneath *Persistence* > *PersistentVolumeClaim* > *Claim Name* remove `central-db`.

Reviewer 3: ```suggestion
. For *Persistence* -> *PersistentVolumeClaim* -> *Claim Name*, remove `central-db`.
```

> . If necessary, a Certificate Authority can be specified in order for Central to trust the database certificate. This can be added by navigating to the YAML view and adding a TLS block underneath the top-level spec:

Reviewer 3: ```suggestion
. If necessary, you can specify a Certificate Authority for Central to trust the database certificate. To add this, navigate to the YAML view and add a TLS block under the top-level spec, as shown in the following example:
```
